### PR TITLE
External config script

### PR DESCRIPTION
**Pull request by mwasilew**
I think I made it working in very basic cases. There are no tests for the script yet. It is likely there are bugs. I didn't test more complex features like adding group with projects inside.

Some operations are not supported for sure:
 - managing UserGroups in Group object
 - moving Projects across Groups
These 2 items would only be possible if object IDs were saved in the config file.

Work remaining to be done:
 - support multiple config files (right now config has to be stored in single file)
 - support external files for things like text input (to improve text formatting)

If the current approach is too complex (I think it is) we should consider other options. For example managing simple objects without any nested complex objects is fairly easy. Storing object IDs would make it way easier to work with. Right now this is WIP but it's already functional.

### CODE REVIEW

**Comment by mwasilew** [comment]
Ah, I forgot it depends on #261

File: scripts/squad-config [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+        self.url_scheme = urlparts.scheme
+
+        connection_token = "Token %s" % self.__get_connection_token(args.url)
+        self.headers = {
+            "Authorization": connection_token
+        }
+        self.old_config = None  # SquadConfig()
+        self.old_backend_config = {}
+
+    def __get_connection_token(self, url):
```
Comment (terceiro) [inline scripts/squad-config]: the usual convention for private methods is to also finish the method name with two underscores (`__`)

(ditto for the several other private methods)

File: scripts/squad-config [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+                        "--save-to-file",
+                        default=None,
+                        help="Save current config to file",
+                        dest="save_filename")
+    parser.add_argument("-y",
+                        "--assume-yes",
+                        action="store_true",
+                        default=False,
+                        help="Assume 'yes' answer for all questions",
+                        dest="assume_yes")
```
Comment (terceiro) [inline scripts/squad-config]: this option is not used for anything

File: scripts/squad-config [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c",
+                        "--config-files",
+                        nargs="+",
+                        default=[],
+                        help="Instance config files",
+                        dest="config_files")
```
[the remark this answers is not in the document]
Comment (mwasilew) [reply scripts/squad-config]: hmm, I don't think it's the case. You pass
--config-files file1.yaml file2.yaml file3.yaml ...
At least it works for me this way.

File: scripts/squad-config [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+    parser.add_argument("-v",
+                        "--debug",
+                        action="store_true",
+                        default=False,
+                        help="Enable debug",
+                        dest="debug")
+    parser.add_argument("-u",
+                        "--url",
+                        help="SQUAD instance URL",
+                        dest="url")
```
[the remark this answers is not in the document]
Comment (mwasilew) [reply scripts/squad-config]: +1

File: scripts/squad-config (hunk 1 above)
Comment (mwasilew) [reply scripts/squad-config]: ACK

File: scripts/squad-config (hunk 2 above)
Comment (mwasilew) [reply scripts/squad-config]: Arhg, it was used initially. It's required for using the script in non-interactive mode. I'll fix it.

**Comment by mwasilew** [comment]
I moved the config script to use flat list instead of nested object. This makes the implementation a bit easier. I tested locally with various scenarios - adding, modifying objects, missing references, etc. There are probably a few scenarios that were not tested, but it should be OK (when used carefully). 

There is no object caching for the moment and there is no check for internal config file integrity. At least the latter should be added in order to make the script immune to typing mistakes. 

Also support for "include" clause was added. This should eliminate quite a few duplicate entries in the config. I think some docs are needed before merging. Any comments are welcome.

**Comment by mwasilew** [comment]
I updated commit message as I didn't want to deal with Environments yet

File: scripts/squad-config [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+        return req.prepare()
+
+    def __get_connection_token__(self, url):
+        netrcauth = netrc.netrc()
+        try:
+            self.username, _, self.token = netrcauth.authenticators(self.base_url)
+            logger.info("Using username: %s" % self.username)
+            return self.token
+        except TypeError:
+            logger.error("No credentials found for %s" % self.base_url)
```
Comment (terceiro) [inline scripts/squad-config]: it should exit here; if a first attempt to get credentials don't work, there is no point in continuing

**Comment by mwasilew** [comment]
Creating many objects at the same time is a problem. Could you send me the example of what you tried to create. Maybe it can be fixed.

File: scripts/squad-config (hunk 5 above)
Comment (mwasilew) [reply scripts/squad-config]: ACK

**Comment by terceiro** [comment]
@mwasilew I don't have it anymore but it was something simple like:

```
- usergroup: foo
  attributes:
    name: foo

- group: foo
  attributes:
    name: foo
    slug: foo
    user_groups:
      - foo

- project: foo-bar
  attributes:
    group: foo
    slug: bar
    name: Bar
```

**Comment by mwasilew** [comment]
I tried this example and it sort of worked. I can do group + usergroup but not all 3 objects. Exception is different, so I need to fix that first. 

There is a slight problem with project. There is no default set for emailtemplate. Since there is no emailtemplate set in the project settings you get 'too many objects found'. That is either a problem with API or the script tries to match None to some object. I'll try to fix it.

**Comment by terceiro** [comment]
@mwasilew thanks. I guess we can go ahead with this initial version?

**Comment by mwasilew** [comment]
@terceiro yep, let's try.

**Pull request merged**